### PR TITLE
USHIFT-4268: Drop `src` from microshift jobs

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance periodic-ci-openshift-microshift-release-4.18-periodics-footprint-and-performance-nightly

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance periodic-ci-openshift-microshift-release-4.18-periodics-footprint-and-performance-nightly pull-ci-openshift-microshift-release-4.17-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance periodic-ci-openshift-microshift-release-4.18-periodics-footprint-and-performance-nightly pull-ci-openshift-microshift-release-4.17-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.17-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance periodic-ci-openshift-microshift-release-4.18-periodics-footprint-and-performance-nightly

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-metal-periodic-test-arm

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-metal-periodic-test

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-microshift pull-ci-openshift-microshift-release-4.14-metal-periodic-test pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test-arm pull-ci-openshift-microshift-release-4.15-test-rpm pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test periodic-ci-openshift-microshift-release-4.16-ocp-conformance-with-optional-rpms-rhel-eus-nightly periodic-ci-openshift-microshift-release-4.15-rebase-on-nightlies

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.16-microshift-metal-tests pull-ci-openshift-microshift-release-4.17-microshift-metal-tests

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-origin-master-e2e-aws-ovn-microshift pull-ci-openshift-microshift-release-4.14-metal-periodic-test pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test-arm pull-ci-openshift-microshift-release-4.15-test-rpm pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test periodic-ci-openshift-microshift-release-4.16-ocp-conformance-with-optional-rpms-rhel-eus-nightly periodic-ci-openshift-microshift-release-4.15-rebase-on-nightlies pull-ci-openshift-microshift-release-4.16-microshift-metal-tests pull-ci-openshift-microshift-release-4.17-microshift-metal-tests

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.16-ocp-metal-nightly pull-ci-openshift-microshift-release-4.16-metal-periodic-test periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.16-ocp-metal-nightly pull-ci-openshift-microshift-release-4.16-metal-periodic-test periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.16-ocp-metal-nightly pull-ci-openshift-microshift-release-4.16-metal-periodic-test periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.16-ocp-metal-nightly pull-ci-openshift-microshift-release-4.16-metal-periodic-test periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance pull-ci-openshift-origin-master-e2e-aws-ovn-microshift pull-ci-openshift-microshift-release-4.14-metal-periodic-test pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test-arm pull-ci-openshift-microshift-release-4.15-test-rpm pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test periodic-ci-openshift-microshift-release-4.16-ocp-conformance-with-optional-rpms-rhel-eus-nightly periodic-ci-openshift-microshift-release-4.15-rebase-on-nightlies pull-ci-openshift-microshift-release-4.16-microshift-metal-tests pull-ci-openshift-microshift-release-4.17-microshift-metal-tests

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-footprint-and-performance pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.16-ocp-metal-nightly pull-ci-openshift-microshift-release-4.16-metal-periodic-test periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance pull-ci-openshift-origin-master-e2e-aws-ovn-microshift pull-ci-openshift-microshift-release-4.14-metal-periodic-test pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test-arm pull-ci-openshift-microshift-release-4.15-test-rpm pull-ci-openshift-microshift-release-4.17-footprint-and-performance pull-ci-openshift-microshift-main-metal-bootc-test periodic-ci-openshift-microshift-release-4.16-ocp-conformance-with-optional-rpms-rhel-eus-nightly periodic-ci-openshift-microshift-release-4.15-rebase-on-nightlies pull-ci-openshift-microshift-release-4.16-microshift-metal-tests pull-ci-openshift-microshift-release-4.17-microshift-metal-tests

File: ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh [hunk 1]
```diff
@@ -128,6 +128,34 @@ function download_microshift_scripts() {
...
 }
 
+function clone_src() {
+    if [ -z ${CLONEREFS_OPTIONS+x} ]; then
+        # Without `src` build, there's no CLONEREFS_OPTIONS, but it can be assembled from $JOB_SPEC
+        CLONEREFS_OPTIONS=$(echo "${JOB_SPEC}" | jq '{"src_root": "/go", "log":"/dev/null", "git_user_name": "ci-robot", "git_user_email": "ci-robot@openshift.io", "fail": true, "refs": [.refs, .extra_refs[]]}')
+        export CLONEREFS_OPTIONS
+    fi
+
+    go_version=$(go version | awk '{print $3}' | tr -d '[a-z]' | cut -f2 -d.)
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh:138]:
```suggestion
    local -r go_version=$(go version | awk '{print $3}' | tr -d '[a-z]' | cut -f2 -d.)
```

File: ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh [hunk 2]
```diff
@@ -128,6 +128,34 @@ function download_microshift_scripts() {
...
+        git remote add origin https://github.com/kubernetes-sigs/prow.git
+        git fetch origin 1a7a18f054ada0ed638678c1ee742ecfc9742958
+        git reset --hard FETCH_HEAD
+    else
+        git clone --depth 1 https://github.com/kubernetes-sigs/prow.git /tmp/prow
+        cd /tmp/prow
+    fi
+    go build -mod=mod -o /tmp/clonerefs ./cmd/clonerefs
+
+    umask 0002
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh:154]: Can we add a comment on what we're trying to do here?

File: ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh [hunk 3]
```diff
@@ -128,6 +128,34 @@ function download_microshift_scripts() {
     chmod 755 "${DNF_RETRY}"
 }
 
+function clone_src() {
+    if [ -z ${CLONEREFS_OPTIONS+x} ]; then
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/includes/openshift-microshift-includes-commands.sh:132]: Can we move this block towards the end right before it's used?

File: ci-operator/step-registry/openshift/microshift/infra/conformance-setup/openshift-microshift-infra-conformance-setup-commands.sh [hunk 4]
```diff
@@ -13,6 +13,8 @@ if "${SRC_FROM_GIT}"; then
     branch="main"
   fi
   git clone https://github.com/openshift/microshift -b $branch /go/src/github.com/openshift/microshift
+else
+  clone_src
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/infra/conformance-setup/openshift-microshift-infra-conformance-setup-commands.sh:17]: Can we name this function with `ci_` prefix?

**Comment by pmtk** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.14-microshift-metal-tests pull-ci-openshift-microshift-release-4.15-metal-periodic-test pull-ci-openshift-microshift-main-metal-bootc-test pull-ci-openshift-microshift-release-4.17-metal-bootc-periodic-test pull-ci-openshift-microshift-main-ocp-conformance-rhel-eus pull-ci-openshift-origin-release-4.18-e2e-aws-ovn-microshift periodic-ci-openshift-microshift-release-4.17-e2e-aws-ovn-ocp-conformance pull-ci-openshift-microshift-main-metal-bootc-test

**Review by pacevedom — APPROVED** [review]
/lgtm

**Pull request merged**